# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientManager.cpp
# Commit: 4ae29776c6b9
# Full Hash: 4ae29776c6b914009f87ce4f549a13d737bfe047
# Author: Luca Greco <lgreco@mozilla.com>
# Date: 2021-09-07 09:48:49
# Description:
#   Bug 1729123 - Run diagnostic assertion on cm->IsShutdown only if dom.workers.testing.enabled is true. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D124687
# ==============================================================================

diff -r 04e64eeedc05 -r 4ae29776c6b9 dom/clients/manager/ClientManager.cpp
--- a/dom/clients/manager/ClientManager.cpp	Mon Sep 06 18:56:11 2021 +0000
+++ b/dom/clients/manager/ClientManager.cpp	Mon Sep 06 19:36:25 2021 +0000
@@ -14,6 +14,7 @@
 #include "mozilla/ipc/BackgroundChild.h"
 #include "mozilla/ipc/PBackgroundChild.h"
 #include "mozilla/ClearOnShutdown.h"  // PastShutdownPhase
+#include "mozilla/StaticPrefs_dom.h"
 #include "nsContentUtils.h"
 #include "prthread.h"
 
@@ -218,12 +219,15 @@
   }
 
   MOZ_DIAGNOSTIC_ASSERT(cm);
-  // Check that the ClientManager instance associated to the current thread has
-  // not been kept alive when it was expected to have been already deallocated
-  // (e.g. due to a leak ClientManager's mShutdown can have ben set to true from
-  // its RevokeActor method but never fully deallocated and unset from the
-  // thread locals).
-  MOZ_DIAGNOSTIC_ASSERT(!cm->IsShutdown());
+
+  if (StaticPrefs::dom_workers_testing_enabled()) {
+    // Check that the ClientManager instance associated to the current thread
+    // has not been kept alive when it was expected to have been already
+    // deallocated (e.g. due to a leak ClientManager's mShutdown can have ben
+    // set to true from its RevokeActor method but never fully deallocated and
+    // unset from the thread locals).
+    MOZ_DIAGNOSTIC_ASSERT(!cm->IsShutdown());
+  }
   return cm.forget();
 }
 
